projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/secrets/service/vault/TestVaultJobSecretsServiceIT.java

```diff
@@ -37,19 +45,73 @@ public class TestVaultJobSecretsServiceIT extends BaseIT {
 
   @Container
   private static final VaultContainer vaultContainer =
-      new VaultContainer<>("vault:1.0.2").withVaultToken("root");
+      new VaultContainer<>("vault:1.13.3").withVaultToken("root");
 
   private static VaultJobSecretsService vaultJobSecretService;
 
   @BeforeAll
-  public static void init() throws URISyntaxException {
+  public static void init() throws URISyntaxException, IOException, InterruptedException {
     String vaultUri = vaultContainer.getHttpHostAddress();
 
-    VaultEndpoint vaultEndpoint = VaultEndpoint.from(new URI(vaultUri));
-    TokenAuthentication clientAuthentication = new TokenAuthentication("root");
+    // Setup vault app roles authentication
+    // https://developer.hashicorp.com/vault/tutorials/auth-methods/approle
+    // enable AppRoles
+    vaultContainer.execInContainer("vault", "auth", "enable", "approle");
+
+    // Create a new test policy via rest as there's no good way to do it via the command mechanism
+    HttpClient httpClient = HttpClients.createDefault();
+    HttpPost httpPost = new HttpPost(vaultUri + "/v1/sys/policies/acl/testpolicy");
+    httpPost.setHeader("X-Vault-Token", "root");
+    StringEntity requestBody =
+        new StringEntity(
+            "{\n"
+                + "  \"policy\": \"path \\\"secret/*\\\" {\\n"
+                + "  capabilities = [ \\\"create\\\", \\\"read\\\",\\\"update\\\", \\\"patch\\\","
+                + " \\\"delete\\\",\\\"list\\\" ]\\n"
+                + "}\"\n"
+                + "}");
+    httpPost.setEntity(requestBody);
+    httpClient.execute(httpPost);
+
+    // create "test" role with the policy
```

Comment:
I would recommend splitting into differnet methods along the lines of some your comments with similar names as the comments

I would also move this to the VaultUtil or VaultTestSetup class and implement it there and use it here as 1 to 3 lines. 

The test  class would become easier to read, The VautlTestSetup class would follow single responsibility principle and is easier to understand as well, it's a win win .

Reply:
Done.